overlord/snapstate/handlers.go

```diff
@@ -288,21 +294,40 @@ func (m *SnapManager) installPrereqs(t *state.Task, base string, prereq []string
 		return err
 	}
 
+	// on systems without core or snapd need to install snapd to
+	// make interfaces work - LP: 1819318
+	var tsSnapd *state.TaskSet
+	if base != "core" && !snapdSnapInstalled(st) && !coreSnapInstalled(st) {
+		onInFlightErr := &state.Retry{After: prerequisitesRetryTimeout}
```

Comment:
I think this var with same name in the surrounding scope has already that value so it could be just used